jetpack-start/wp-cli-keys.php

```diff
@@ -0,0 +1,196 @@
+	 *
+	 * ## OPTIONS
+	 *
+	 * --akismet_key=<akismet_key>
+	 * : The Akismet API key provided by the Jetpack Start API
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp jetpack-start keys akismet --akismet_key=12345
+	 *
+	 * @subcommand akismet
+	 */
+	public function install_akismet_keys( $args, $assoc_args ) {
+		if ( ! class_exists( 'Akismet' ) ) {
+			WP_CLI::error( 'Failed to install Akismet keys; Please activate the Akismet plugin and re-run this script.' );
+		}
+
+		if ( defined( 'WPCOM_API_KEY' ) && ! empty( WPCOM_API_KEY ) ) {
+			WP_CLI::error( 'Skipping Akismet install; WPCOM_API_KEY is defined which will always take preference.' );
+		}
+
+		WP_CLI::line( 'Configuring Akismet' );
+
+		$akismet_key = $assoc_args['akismet_key'];
+		$result = $this->register_akismet( $akismet_key );
+		if ( is_wp_error( $result ) ) {
+			WP_CLI::error( '- Failed to configure Akismet: ' . $result->get_error_message() );
+		}
+
+		WP_CLI::success( 'Akismet configured successfully.' );
+	}
+
+	private function register_akismet( $key ) {
+		update_option( 'wordpress_api_key', $key );
+
+		$verification_response = Akismet::verify_key( $key );
+		if ( $verification_response !== 'valid' ) {
+			return new WP_Error( 'jp-start-akismet', 'Akismet key not valid: ' . $verification_response );
+		}
+
+		return true;
+	}
+
+	/**
+	 * Install VaultPress keys
+	 *
+	 * ## OPTIONS
+	 *
+	 * --vaultpress_key=<vaultpress_key>
+	 * : The VaultPress API key provided by the Jetpack Start API.
+	 *
+	 * [--vaultpress_ssh_key=<vaultpress_ssh_key>]
+	 * : The VaultPress SSH key provided by the Jetpack Start API.
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     wp jetpack-start vaultpress --vaultpress_key=12345
+	 *     wp jetpack-start vaultpress --vaultpress_key=12345 --vaultpress_ssh_key="asdfgh!!!"
+	 *
+	 * @subcommand vaultpress
+	 */
+	public function install_vaultpress_keys( $args, $assoc_args ) {
+		if ( ! class_exists( 'VaultPress' ) ) {
+			WP_CLI::error( 'Failed to install VaultPress keys; Please activate the VaultPress plugin and re-run this script.' );
+		}
+
+		WP_CLI::line( 'Configuring VaultPress' );
+
+		$vaultpress_api_key = $assoc_args['vaultpress_key'];
+		$result = $this->register_vaultpress( $vaultpress_api_key );
+		if ( is_wp_error( $result ) ) {
+			WP_CLI::error( '- Failed to configure VaultPress: ' . $result->get_error_message() );
+		}
+
+		$vaultpress_ssh_key = $assoc_args['vaultpress_ssh_key'];
+		if ( $vaultpress_ssh_key ) {
+			$result = $this->install_vaultpress_ssh_keys( $vaultpress_ssh_key );
+			if ( $result ) {
+				WP_CLI::line( '- Added VaultPress SSH key to ' . $result );
+			} else {
+				WP_CLI::warning( '- Failed to find valid, accessible path to add SSH key' );
+			}
+		}
+
+		// TODO: set up WP=>VaultPress SSH access
+
+		WP_CLI::success( 'VaultPress configured successfully.' );
+	}
+
+	private function register_vaultpress( $api_key ) {
+		return VaultPress::register( $api_key );
+	}
+
+	private function install_vaultpress_ssh_keys( $ssh_key ) {
```

Comment:
How is this planned to be used in practice? Obviously we need to take great care with it. Maybe we can leave it off on VIP, since it won't work anyway?